src/replica/python/schema_migration.py

```diff
@@ -80,6 +92,47 @@ def _set_version(self, version):
             raise RuntimeError(
                 f"Failed to update version number in database to {version}, current version is now {current}")
 
+    def _create_database(self):
+        """Create the replication controller database.
+        """
+        with closing(self.connection.cursor()) as cursor:
+            cursor.execute(f"CREATE DATABASE {self.database};")
+            warnings = cursor.fetchwarnings()
+            if warnings:
+                _log.warn("Warnings were creating database %s", self.database)
+        _log.info(f"Created database {self.database}.")
+        self.connection.commit()
+
+    def _create_users(self):
+        """Create the users for the replication controller database.
+        """
+        if not self.repl_connection:
+            raise RuntimeError(
+                "A non-admin replication database connection uri must be provided to initialize the "
+                "replication database."
+            )
+        user = make_url(self.repl_connection).username
+        if not user:
+            raise RuntimeError(
+                "To initialize the replication database, the non-admin connection uri must contain a user "
+                "name."
+            )
+        for stmt in [
+            f"CREATE USER IF NOT EXISTS {user}@localhost;",
+            f"CREATE USER IF NOT EXISTS {user}@'%';",
+            f"CREATE DATABASE IF NOT EXISTS qservReplica;",
```

Comment:
The name of the database is not predefined. It could be different in some test environments as well as in a possible case when the same instance of MySQL would be shared. In this case, one would need the flexibility to name these databases differently.

Reply:
it's an error that this line got left in. (The database is created in the function above this one). I'll remove this line.

Reply:
👍